src/Core/Index/IndexMap.inl

```diff
@@ -301,6 +330,17 @@ inline T& IndexMap< T >::operator[]( const uint i ) {
     return access( i );
 }
 
+template < typename T >
+inline const T& IndexMap< T >::operator[]( const Index& idx ) const {
+    return at( idx );
+}
+
+
+
+template < typename T >
+inline const T& IndexMap< T >::operator[]( const uint i ) const{
+    return at( i );
+}

```

Comment:
Why do we need a const uint version of this function ?
I sounds better to me to stick to Index, by adding an implicit constructor or cast operator.
By this mean, we know that Index will always be used, without any overhead, which is better for testing and monitoring.